src/firefly/js/fieldGroup/FieldGroupCntlr.js

```diff
@@ -386,8 +386,7 @@ const updateFieldGroupMount= function(state,action) {
 const fireFieldsReducer= function(fg, action) {
     // return  fg.reducerFunc ? fg.reducerFunc(revalidateFields(fg.fields), action) : fg.fields;
     if (fg.reducerFunc ) {
-        const newFields= fg.reducerFunc(revalidateFields(fg.fields), action);
-        return smartMerge(fg.fields,newFields);
+        return fg.reducerFunc(revalidateFields(fg.fields), action);
```

Comment:
With this change, we are reliant on the reducerFunc to do the right thing, which returns the original `fields` if nothing has changed, or returns a new Immutable-styled updated `fields` object where there are changes.
That said, I found couple of places where this may break the existing behavior.
In firefly:
 src/firefly/js/ui/CatalogSearchMethodType.jsx#searchMethodTypeReducer:437
This function returns new object regardless whether or not there were changes.  

In irsa-ife:
src/irsa/js/finderchart/FinderchartDownloadDialog.jsx#FinderChartOptionDLReducer:380
This function returns new object regardless whether or not there were changes.